pubsub/google/cloud/pubsub_v1/publisher/_batch/thread.py

```diff
@@ -292,12 +300,21 @@ def publish(self, message):
             if not self.will_accept(message):
                 return future
 
-            new_size = self._size + message.ByteSize()
+            size_increase = types.PublishRequest(messages=[message]).ByteSize()
+
+            if (self._base_request_size + size_increase) > _SERVER_PUBLISH_MAX_BYTES:
+                err_msg = (
+                    "The message being published would produce too large a publish "
+                    "request that would exceed the maximum allowed size on the "
+                    "backend ({} bytes).".format(_SERVER_PUBLISH_MAX_BYTES)
+                )
+                raise ValueError(err_msg)
+
+            new_size = self._size + size_increase
             new_count = len(self._messages) + 1
-            overflow = (
-                new_size > self.settings.max_bytes
-                or new_count >= self._settings.max_messages
-            )
+
+            size_limit = min(self.settings.max_bytes, _SERVER_PUBLISH_MAX_BYTES)
+            overflow = new_size > size_limit or new_count >= self._settings.max_messages
```

Comment:
I'd just like to take a minute to register my disgust at having both `self.settings` and `self._settings`.

Reply:
I can fix that, too, along the way, some kind of a relic of the past.

Reply:
Since this code is under change anyway, can you change `self._settings` to `self.settings`?

Reply:
Yes.
I.
Can.
:+1:

Reply:
Thanks!